front-end/src/app/shared/models/f3x-summary.model.ts

```diff
@@ -0,0 +1,247 @@
+import { BaseModel } from './base.model';
+import { Report } from '../interfaces/report.interface';
+import { plainToClass } from 'class-transformer';
+import { LabelList } from '../utils/label.utils';
+
+export enum F3xFormTypes {
+  F3XN = 'F3XN',
+  F3XA = 'F3XA',
+  F3XT = 'F3XT',
+}
+
+export type F3xFormType = F3xFormTypes.F3XN | F3xFormTypes.F3XA | F3xFormTypes.F3XT;
+
+export const F3xFormTypeLabels: LabelList = [
+  [F3xFormTypes.F3XN, 'FORM 3X'],
+  [F3xFormTypes.F3XA, 'FORM 3X'],
+  [F3xFormTypes.F3XT, 'FORM 3X'],
+];
+
+export const F3xFormVersionLabels: LabelList = [
+  [F3xFormTypes.F3XN, 'Original'],
+  [F3xFormTypes.F3XA, 'Amendment'],
+  [F3xFormTypes.F3XT, 'Termination'],
+];
+
+export enum F3xReportCodes {
+  Q1 = 'Q1',
+  Q2 = 'Q2',
+  Q3 = 'Q3',
+  YE = 'YE',
+  TER = 'TER',
+  MY = 'MY',
+  TwelveG = '12G',
+  TwelveP = '12P',
+  TwelveR = '12R',
+  TwelveS = '12S',
+  TwelveC = '12C',
+  ThirtyG = '30G',
+  ThirtyR = '30R',
+  ThirtyS = '30S',
+  M2 = 'M2',
+  M3 = 'M3',
+  M4 = 'M4',
+  M5 = 'M5',
+  M6 = 'M6',
+  M7 = 'M7',
+  M8 = 'M8',
+  M9 = 'M9',
+  M10 = 'M10',
+  M11 = 'M11',
+  M12 = 'M12',
+}
+
+export type F3xReportCode =
+  | F3xReportCodes.Q1
+  | F3xReportCodes.Q2
+  | F3xReportCodes.Q3
+  | F3xReportCodes.YE
+  | F3xReportCodes.TER
+  | F3xReportCodes.MY
+  | F3xReportCodes.TwelveG
+  | F3xReportCodes.TwelveP
+  | F3xReportCodes.TwelveR
+  | F3xReportCodes.TwelveS
+  | F3xReportCodes.TwelveC
+  | F3xReportCodes.ThirtyG
+  | F3xReportCodes.ThirtyR
+  | F3xReportCodes.ThirtyS
+  | F3xReportCodes.M2
+  | F3xReportCodes.M3
+  | F3xReportCodes.M4
+  | F3xReportCodes.M5
+  | F3xReportCodes.M6
+  | F3xReportCodes.M7
+  | F3xReportCodes.M8
+  | F3xReportCodes.M9
+  | F3xReportCodes.M10
+  | F3xReportCodes.M11
+  | F3xReportCodes.M12;
+
+// NOTE: If this label list is updated, please sync up the values in the api backend as well:
+// https://github.com/fecgov/fecfile-web-api/blob/develop/django-backend/fecfiler/f3x_summaries/views.py
```

Comment:
this could be more lift than it's worth, could we just have this mapping live in the f3x view and provide an endpoint to retrieve it?

Reply:
OR have it live here and pass the mapping in the list request. I just want to avoid the duplicate definition

Reply:
The label values are stable enough that I'd rather not have an endpoint lookup. I did look into passing the alias field, report_code_label, back to the client as part of the result set but did not find a way to do that readily available. We can break that task out as a separate ticket as I should think it possible in the serializer somehow